spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -606,6 +607,29 @@ public View createView(
   @Override
   public View alterView(Identifier ident, ViewChange... changes)
       throws NoSuchViewException, IllegalArgumentException {
+    if (null != asViewCatalog) {
+      try {
+        org.apache.iceberg.view.View view = asViewCatalog.loadView(buildIdentifier(ident));
+        UpdateViewProperties updateViewProperties = view.updateProperties();
```

Comment:
Can you also reject reserved properties? I don't think that you should be able to set `format-version`, `location`, or `provider`.

Reply:
this is a miss on my end. For some reason reason I manually checked `provider` / `location` (which Spark would prevent from being set/unset) and assumed Spark would do this for all reserved properties.

I've added some tests and fixed this. Note that I'm rejecting both setting/removing reserved properties as it seems a bit confusing from a user's perspective if Spark rejects setting/removing certain properties but we only reject setting those.
It also appears that the behavior for tables is slightly different, where we're only rejecting setting reserved  table properties, but we're not rejecting the removal of reserved  table properties.